Classes/Private/Manager/Submanagers/OCTSubmanagerCalls.m

```diff
@@ -246,11 +236,45 @@ - (void)updateCall:(OCTCall *)call withStatus:(OCTCallStatus)status
     OCTRealmManager *realmManager = [self.dataSource managerGetRealmManager];
 
     [realmManager updateObject:call withBlock:^(OCTCall *callToUpdate) {
-
         callToUpdate.status = status;
+    }];
+}
+
+- (void)pause:(BOOL)pause controlSentToCall:(OCTCall *)call
+{
+    OCTRealmManager *realmManager = [self.dataSource managerGetRealmManager];
+
+    BOOL wasPaused = call.pausedStatus != OCTCallPausedStatusNone;
+    OCTCallPausedStatus previousPausedStatus = call.pausedStatus;
 
-        callToUpdate.onHoldStartInterval = (status == OCTCallStatusPaused) ? [[NSDate date] timeIntervalSince1970] : 0;
+    [realmManager updateObject:call withBlock:^(OCTCall *callToUpdate) {
+
+        if (pause && ! wasPaused) {
+            callToUpdate.onHoldStartInterval = [[NSDate date] timeIntervalSince1970];
+        }
+
+        if (pause) {
+            callToUpdate.pausedStatus |= OCTCallPausedStatusByUser;
+        }
+        else {
+            callToUpdate.pausedStatus &= ~OCTCallPausedStatusByUser;
+        }
     }];
+
+    OCTFriend *friend = [call.chat.friends firstObject];
+
+    if (pause && ! wasPaused) {
+        [self.timer stopTimer];
+        [self.audioEngine stopAudioFlow:nil];
+    }
+    else if (! pause &&  (previousPausedStatus == OCTCallPausedStatusByUser)) {
+        [self.audioEngine startAudioFlow:nil];
+        [self.timer startTimerForCall:call];
+        self.audioEngine.friendNumber = friend.friendNumber;
+    }
+    else if (! pause) {
+        self.audioEngine.friendNumber = friend.friendNumber;
+    }
```

Comment:
>
Also replaced previousPausedStatus == OCTCallPausedStatusByUser with previousPausedStatus & OCTCallPausedStatusByUser. I think it should be there, right?

Hmm, no you can't do that there. You want to check that *only* `OCTCallPausedStatusByUser` is part of the bitmask, so you know whether or not to start the audioEngine back up again.